lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -77,67 +103,188 @@ public PausedDuration pause(int pauseDuration, @Nullable Long maxPauseWaitTime)
     }
 
-    TargetSnapshot target;
+    // Get pods and deployment information before pause.
+    TargetSnapshot targetBeforePause;
     try {
-      target = targetSelector.select();
+      targetBeforePause = getTarget();
     } catch (Exception e) {
       throw new PauserException("Failed to find the target pods to pause.", e);
     }
 
-    RequestCoordinator coordinator = getRequestCoordinator(target);
-
-    coordinator.pause(true, maxPauseWaitTime);
-
-    Instant startTime = Instant.now();
+    // Get RequestCoordinator of Scalar Admin to pause.
+    RequestCoordinator requestCoordinator;
+    try {
+      requestCoordinator = getRequestCoordinator(targetBeforePause);
+    } catch (Exception e) {
+      throw new PauserException("Failed to initialize the request coordinator.", e);
+    }
 
-    Uninterruptibles.sleepUninterruptibly(pauseDuration, TimeUnit.MILLISECONDS);
+    // From here, we cannot throw exceptions right after they occur because we need to take care of
+    // the unpause operation failure. We will throw the exception after the unpause operation or at
+    // the end of this method.
 
-    Instant endTime = Instant.now();
+    // Run a pause operation.
+    PausedDuration pausedDuration = null;
+    PauseFailedException pauseFailedException = null;
+    try {
+      pausedDuration = pauseInternal(requestCoordinator, pauseDuration, maxPauseWaitTime);
+    } catch (Exception e) {
+      pauseFailedException = new PauseFailedException(PAUSE_ERROR_MESSAGE, e);
+    }
 
-    unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
+    // Run an unpause operation.
+    UnpauseFailedException unpauseFailedException = null;
+    try {
+      unpauseWithRetry(requestCoordinator, MAX_UNPAUSE_RETRY_COUNT);
+    } catch (Exception e) {
+      unpauseFailedException = new UnpauseFailedException(UNPAUSE_ERROR_MESSAGE, e);
+    }
 
-    TargetSnapshot targetAfterPause;
+    // Get pods and deployment information after pause.
+    TargetSnapshot targetAfterPause = null;
+    GetTargetAfterPauseFailedException getTargetAfterPauseFailedException = null;
     try {
-      targetAfterPause = targetSelector.select();
+      targetAfterPause = getTarget();
     } catch (Exception e) {
-      throw new PauserException(
-          "Failed to find the target pods to examine if the targets pods were updated during"
-              + " paused.",
-          e);
+      getTargetAfterPauseFailedException =
+          new GetTargetAfterPauseFailedException(GET_TARGET_AFTER_PAUSE_ERROR_MESSAGE, e);
     }
 
-    if (!target.getStatus().equals(targetAfterPause.getStatus())) {
-      throw new PauserException("The target pods were updated during paused. Please retry.");
+    // Check if pods and deployment information are the same between before pause and after pause.
+    StatusCheckFailedException statusCheckFailedException = null;
+    StatusUnmatchedException statusUnmatchedException = null;
+    try {
```

Comment:
How about executing this block only when `targetAfterPause != null`? The method seems to return `StatusUnmatchedException` when the argument `after` is null, but in this case comparing `before` and `after` itself isn't conducted.

Reply:
Ah, yes. That's right. I will fix it. Thank you for the suggestion!

Reply:
Fixed in 5cca44c080bffc253678d0dcb89ba3c9ec432d11.